AdminSetPassword.php

```diff
@@ -41,24 +41,40 @@ public function init()
         $this->current_page = basename($_SERVER['PHP_SELF']);
     }
 
+    /**
+     * @return array
+     */
     public function hooks()
     {
         return [
-            'EVENT_LAYOUT_CONTENT_BEGIN' => 'addPageContent'
+            'EVENT_LAYOUT_RESOURCES' => 'enqueueScript',
+            'EVENT_LAYOUT_CONTENT_BEGIN' => 'printTranslations',
```

Comment:
Not really sure why you're hooking a separate event for the translations.

Probably (but I didn't test) everything could go in RESOURCES. Keeping the logic to identify the current page, you would then only load the setpassword.js script when it is actually needed instead of all the time.

Reply:
Changed, now I use one hook.